request_context.go

```diff
@@ -0,0 +1,147 @@
+package requestcontext
+
+import (
+	"encoding/json"
+	"os"
+	"strings"
+
+	"github.com/juju/errgo"
+	"github.com/op/go-logging"
+)
+
+const (
+	separator = " | "
+)
+
+var (
+	levels = map[string]int{
+		"critical": 0,
+		"error":    1,
+		"warning":  2,
+		"notice":   3,
+		"info":     4,
+		"debug":    5,
+	}
+)
+
+type Ctx map[string]interface{}
+
+func (ctx Ctx) isEmpty() bool {
+	return len(ctx) == 0
+}
+
+type Logger struct {
+	config LoggerConfig
+	logger *logging.Logger
+}
+
+type LoggerConfig struct {
+	Name  string
+	Level string
```

Comment:
Since ever we provided this levels as string, as it is easier to pass through CLI. It just reflects the levels provided by the underlying logger: https://godoc.org/github.com/op/go-logging#Level